benchmarking/example_classification.py

```diff
@@ -54,54 +66,45 @@
     X_test, y_test = X[test_idx], y[test_idx]
     X_valid, y_valid = X[valid_idx], y[valid_idx]
 
-    records = []
+    X_train_mols = [MolFromSmiles(smiles) for smiles in X_train]
+    X_valid_mols = [MolFromSmiles(smiles) for smiles in X_valid]
+    X_test_mols = [MolFromSmiles(smiles) for smiles in X_test]
 
-    fp_names = [
-        "ECFP",
-        "Atom Pairs",
-        "Topological Torsion",
-        "MACCS Keys",
-        "ERG",
-        "MAP4",
-        "MHFP",
-    ]
-    fprints = [
-        ECFPFingerprint,
-        AtomPairFingerprint,
-        TopologicalTorsionFingerprint,
-        MACCSKeysFingerprint,
-        ERGFingerprint,
-        MAP4Fingerprint,
-        MHFPFingerprint,
-    ]
-
-    clf_names = ["RF", "LogReg", "LGBM"]
-    classifiers = [
-        RandomForestClassifier,
-        LogisticRegression,
-        lgb.LGBMClassifier,
-    ]
-    classifier_kwargs = [
-        {"n_jobs": -1, "class_weight": "balanced"},
-        {
-            "class_weight": "balanced",
-            "penalty": None,
-            "n_jobs": -1,
-        },
-        {"n_jobs": -1, "class_weight": "balanced", "verbose": 0},
-    ]
+    records = []
 
     np.random.seed(42)
 
-    for fingerprint, fp_name in zip(fprints, fp_names):
+    for fingerprint in fingerprint_classes:
+        fp_name = fingerprint.__name__
         records.append({})
         records[-1]["fp_name"] = fp_name
-        print(fp_name, "Fingerprint")
+        print(fp_name)
         start = time()
         fp_transformer = fingerprint(n_jobs=-1)
-        X_fp_train = fp_transformer.transform(X_train)
-        X_fp_valid = fp_transformer.transform(X_valid)
-        X_fp_test = fp_transformer.transform(X_test)
+
+        try:
+            X_fp_train = fp_transformer.fit_transform(X_train)
+            X_fp_valid = fp_transformer.transform(X_valid)
+            X_fp_test = fp_transformer.transform(X_test)
+        except RuntimeError as e:
+            print(f" - Error: {e}")
+            records[-1]["error"] = str(e)
+            continue
+        except ValueError as e:
+            try:
+                conf_gen = ConformerGenerator()
+                X_train_conf = conf_gen.transform(X_train_mols)
+                X_valid_conf = conf_gen.transform(X_valid_mols)
+                X_test_conf = conf_gen.transform(X_test_mols)
```

Comment:
We should generate conformers once and then use them as needed. Recomputing them will be very expensive. Maybe we could introduce an attribute to indicate which fingerprints require conformers, that would get rid of this try/except

Reply:
The attribute would be very helpful